### DISCUSSION TITLE
COLang 2.0

### DISCUSSION THREAD

**Discussion by jingnanzhou** [NVIDIA-NeMo/Guardrails#448] [General]
@drazvan , Does Colang 2.0 work in v0.82?  I tried to run hello world by using CoLang 2.0 in v0.82,  I got error message: "No LLM calls were made".  when v0.90 will be released?

**Comment by drazvan** [comment]
Thanks for the interest @jingnanzhou. Version 0.9.0 will be published at the end of this month, including a whole new documentation section around it.

**Reply by jingnanzhou** [reply]
@drazvan , I have spent a lot of time on digging into v2.0 code base. Is there any way I could participate and contribute?

**Reply by drazvan** [reply]
We've not yet added any issues related to Colang 2.0. We'll start adding some next week, and I'll tag you. Thanks!

**Comment by d-shree** [comment]
@drazvan  - Is there an estimated timeline for when the stable version of Colang 2.0 will be released? Also will Colang 1.0 be supported for long post the release of a stable Colang 2.0 version?

**Reply by drazvan** [reply]
We expect Colang 2.0 to be in Beta for roughly the next 3 months. So, for 0.10.0 and 0.11.0 it will still be Beta. After that, Colang 1.0 will be deprecated. We don't have an exact timeline for how long it will still be supported. We will keep compatibility for sure for a little longer. At the same time, we'll try to make the migration process as smooth as possible, and all the new features will target Colang 2.0. There shouldn't be any strong reason to stick with 1.0 (apart from the effort to adapt the syntax, for which we'll try to provide a converter that will do 90% of the job).
